activerecord/lib/active_record/insert_all.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class InsertAll
+    def initialize(model, inserts, options)
+      @model = model
+      @inserts = inserts
+      @options = options
+    end
+
+    def execute
+      return ActiveRecord::Result.new([], []) if inserts.empty?
+
+      connection.exec_query to_sql, "Bulk Insert"
```

Comment:
This isn't an exceptional state, so having a precondition is a bit odd. We can just do:

```ruby
if inserts.present?
  connection.exec_query to_sql, "Bulk Insert"
else
  ActiveRecord::Result.new([], [])
end
```